src/core/agent/services/credentialService.ts

```diff
@@ -628,26 +630,32 @@ class CredentialService extends AgentService {
         };
       })
     );
-    unreadGrantNotes = unreadGrantNotes.filter(
-      (notification: any) =>
-        notification.exchange.exn.i === serverAid &&
-        notification.exchange.exn.e.acdc &&
-        notification.exchange.exn.e.acdc === expectedDomain
+
+    const matchingUnreadGrants = unreadGrantExnMsgs.filter(
+      (message: { notiId: string; notiSaid: string; exchange: any }) => {
+        return (
+          message.exchange.exn.i === serverAid &&
+          new URL(message.exchange.exn.e.acdc?.a?.domain).hostname ===
```

Comment:
As I mentioned this should be reverted since `domain` in the ACDC is already the hostname value. So this throws an error now since just the hostname is not a valid URL

Reply:
done!